### PR TITLE
feat(authors): add authors data to schemas

### PR DESCRIPTION
**Pull request by jpetto**
## DO NOT MERGE
Until the corresponding changes to the frontend in https://getpocket.atlassian.net/browse/BACK-1427 are ready to go.

## Goal

add authors data to schemas - public and admin API, prisma, and snowplow. 
+ update all the resolvers to take in/return authors.

RESOLVED - **note** we are waiting on a confirmation on if we need to send `sortOrder` to snowplow. there's a TODO in that file in this PR ([slack thread](https://pocket.slack.com/archives/CTLKW5WCF/p1652133701280769?thread_ts=1651704195.064879&cid=CTLKW5WCF)).

## I'd love feedback/perspectives on:
- this is modeled off our work in collections. anything we could do better?

JIRA ticket:
* https://getpocket.atlassian.net/browse/BACK-1423
* https://getpocket.atlassian.net/browse/BACK-1426

### CODE REVIEW

**Comment by nina-py** [comment]
@jpetto, I've added a migration for changes in Prisma schema and, as I was going about trying to fix the tests I veered into updating the resolvers - another ticket, as it turns out! I'm almost there, but a few tests are still erroring out.

File: src/database/mutations/ScheduledItem.ts
Comment (nina-py) [inline src/database/mutations/ScheduledItem.ts:49]: Despite _seemingly_ pulling in `authors` from Prisma, integration tests for all `ScheduledItem`-related mutations are failing to find the `authors` prop on the included `approvedItem`.

File: src/admin/resolvers/mutations/ApprovedItem.ts
Comment (jpetto) [inline src/admin/resolvers/mutations/ApprovedItem.ts:150]: this could be a new function `getApprovedItemByExternalId`? or do we think getting by URL (which is unique!) is fine?

Comment (nina-py) [reply src/admin/resolvers/mutations/ApprovedItem.ts]: I opted for retrieving this via `getApprovedItemByExternalId`, the one that retrieves by URL is used to query when an item is added manually. I personally feel more comfortable using the id value rather than URL :).

File: schema-admin.graphql
Comment (kschelonka) [inline schema-admin.graphql:156]:
```suggestion
    authors: [CorpusItemAuthor!]!
```

?

File: src/admin/resolvers/mutations/ScheduledItem.integration.ts
Comment (kschelonka) [inline src/admin/resolvers/mutations/ScheduledItem.integration.ts:226]: jest has this built in: https://jestjs.io/docs/expect#expectobjectcontainingobject

File: src/admin/resolvers/mutations/ApprovedItem.ts
Comment (kschelonka) [inline src/admin/resolvers/mutations/ApprovedItem.ts:151]: Doesn't the association table have `externalId`?

File: prisma/schema.prisma
Comment (Herraj) [inline prisma/schema.prisma:108]: nit pick: I think this go accidentally moved from line 105

Can be fixed later on!

File: src/admin/resolvers/mutations/ApprovedItem.ts
Comment (kschelonka) [reply src/admin/resolvers/mutations/ApprovedItem.ts:151]: oh nvm... I think from reading the schema, that's another "external" ID for that specific author-item relationship?

I don't know prisma, but I feel like you *must* be able to delete by association... or add the additional FK

File: src/database/queries/ApprovedItem.ts
Comment (Herraj) [inline src/database/queries/ApprovedItem.ts:37]: Do we need to provide the `sortOrder` for **finding** and **deleting** items from the DB? The fact that we save them to the DB with the `asc` `sortOrder`, shouldn't the item be returned in the same sort order?

Unless it is a prisma thing where do find and delete, it needs to match the parameters of how it was saved?

**Review by Herraj — DISMISSED** [review]
I'm tired so couldn't give you a fully focused review -- if you want to wait until tomorrow, I can review this first thing tomorrow morning and merge it.

But anyways, looks good to me 😄

**Review by kschelonka — APPROVED** [review]
seems like you'd get an easy win adding another FK relationship for approvedItem.externalId to authors, but I see no reason to not ship

File: src/config/index.ts
Comment (bassrock) [inline src/config/index.ts:45]: Long term we should see if analytics will let us make this the GIT SHA

File: src/admin/resolvers/mutations/ApprovedItem.ts
Comment (nina-py) [reply src/admin/resolvers/mutations/ApprovedItem.ts:151]: The way this API is built (and also `collection-api`), with extra "external" ids to avoid exposing internal integer ones, means we can't actually delete related records by association until we get the full approved item record from the database that contains the integer primary key. 

It's the admin/internal part of the API so the performance hit of fetching this record an extra time is negligible. I would do it differently if performance was important.

File: src/database/queries/ApprovedItem.ts
Comment (nina-py) [reply src/database/queries/ApprovedItem.ts:37]: @Herraj, we definitely need to provide `sortOrder` when fetching a full `ApprovedItem` record. Even if records are written to a relational database sequentially, there is no guarantee they will be returned in that order unless you specify it explicitly. 

I am not sure about providing `sortOrder` when _deleting_ items. That shouldn't happen anywhere in the codebase.  The closest to this that I can think of is in the `deleteApprovedItem` mutation where we fetch the entire record, including the authors sorted correctly, _before_ we go ahead and hard delete everything - this is done to return the now deleted approved item as the mutation result for consistency with other mutations.

**Review by Herraj — APPROVED** [review]

**Pull request merged**
